src/coreclr/gc/gc.cpp

```diff
@@ -49307,3 +49377,294 @@ void PopulateDacVars(GcDacVars *gcDacVars)
 #endif // MULTIPLE_HEAPS
     gcDacVars->generation_field_offsets = reinterpret_cast<int**>(&generation_field_offsets);
 }
+
+void GCHeap::RefreshMemoryLimit()
+{
+    gc_heap::refresh_memory_limit();
+}
+
+void gc_heap::configure_memory_settings(bool is_initialization, uint32_t& nhp, uint32_t& nhp_from_config, size_t& seg_size_from_config)
+{
+    // If the hard limit is specified, the user is saying even if the process is already
+    // running in a container, use this limit for the GC heap.
+    if (!hard_limit_config_p)
```

Comment:
we should figure out if it's a needed scenario to support refreshing the hardlimit that was already set when the runtime is loaded.